app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -143,4 +141,14 @@ def api_authorization_token
     }]
     User.current.jwt_token!(expiration: 4.hours.to_i, scope: scope)
   end
+
+  def find_organization
+    return Organization.authorized(:view_organizations).find(params['organization_id']) if params['organization_id'].present?
+    User.current.default_organization || User.current.my_organizations.first
+  end
```

Comment:
After looking at both Katello and this PR, it looks like you want to change the default, but have the non-default left intact.

I think splitting the method would result in cleaner code:

```suggestion
  def find_organization
    return Organization.authorized(:view_organizations).find(params['organization_id']) if params['organization_id'].present?
    
    default_organization
  end
  
  def default_organization
    User.current.default_organization || User.current.my_organizations.first
  end
```

This way in Katello you will only need to override the `default_organization` method:
``` ruby
def default_location
  Location.authorized(:view_locations).find_by(name: Setting[:default_location_subscribed_hosts]) || super
end
```

Reply:
I'm not against splitting it into four methods, but I'd like to know if it is not an overengineered solution.

The solution IMO is to expose only `default_location` & `default_organization`, and leaving `find_organization` and `find_location` logic hidden from extending by plugins.